main.tf

```diff
@@ -324,3 +326,53 @@ resource "null_resource" "confirm_network_healthy" {
     }
   }
 }
+
+
+resource "ibm_container_addons" "addons" {
+  count             = var.enable_autoscaling && !(var.disable_public_endpoint) ? 1 : 0
+  cluster           = local.cluster_name
```

Comment:
This is allowing ibm_container_vpc_cluster.cluster[0].name (ie. not the autoscaling_cluster[0].name) to be used if enable_autoscaling=true and ignore_worker_pool_size_changes=false. This indicates that logic around locals `cluster_id` and `cluster_name` is not right. The `cluster_id` issues would pre-date this PR, it has just been highlighted by the introduction of the `cluster_name`